hoverxref/extension.py

```diff
@@ -139,6 +140,57 @@ def setup_sphinx_tabs(app, config):
             app.disconnect(listener_id)
 
 
+def setup_intersphinx(app, config):
+    """
+    Disconnect ``missing-reference`` from ``sphinx.ext.intershinx``.
+
+    As there is no way to hook into the ``missing_referece`` function to add
+    some extra data to the disutils node returned by this function, we
+    disconnect the original listener and add our custom one.
+
+    https://github.com/sphinx-doc/sphinx/blob/53c1dff/sphinx/ext/intersphinx.py
+
+    Note that ``hoverxref.extension`` has to be installed **after** ``sphinx.ext.intersphinx``.
+    """
+    if sphinx.version_info < (3, 0, 0):
+        listeners = list(app.events.listeners.get('missing-reference').items())
+    else:
+        listeners = [
+            (listener.id, listener.handler)
+            for listener in app.events.listeners.get('missing-reference')
+        ]
+    for listener_id, function in listeners:
+        module_name = inspect.getmodule(function).__name__
+        if module_name == 'sphinx.ext.intersphinx':
+            app.disconnect(listener_id)
+
+
+def missing_reference(app, env, node, contnode):
+    """
+    Override original ``missing_referece`` to add data into the node.
+
+    We call the original intersphinx extension and add hoverxref CSS classes
+    plus the ``data-url`` to the node returned from it.
+    """
+    newnode = sphinx_missing_reference(app, env, node, contnode)
+    intersphinx_enabled = app.config.hoverxref_intersphinx or app.config.hoverxref_auto_ref
+    hoverxref_type = app.config.hoverxref_intersphinx_type or app.config.hoverxref_default_type
```

Comment:
Should we check for this prior to the disconnect, and at the top of this function? I worry we might cause more issues by futzing with intersphinx even when it isn't enabled.

Reply:
Good point! We should check first if it's installed and connected. I think I've used this technique already somewhere else, so it makes sense to do the same here.